docs/source/distributed.rst

```diff
@@ -36,6 +36,35 @@ MPI supports cuda only iff the implementation used to build PyTorch supports it.
 | barrier    | ✓   | ✘   | ✓   | ✓   | ✓   | ?   |
 +------------+-----+-----+-----+-----+-----+-----+
 
+.. _distributed-basics:
+
+Basics
+------
+
+The `torch.distributed` package provides PyTorch support and communication primitives
+for multiprocess parallelism across several computation nodes running on one or more
+machines. The class :func:`torch.nn.parallel.DistributedDataParallel` builds on this
+functionality to provide synchronous distributed training as a wrapper around any
+PyTorch model. This differs from the kinds of parallelism provided by
+:module:`torch.multiprocessing` and :func:`torch.nn.DataParallel` in that it supports
+multiple network-connected machines and in that the user must explicitly launch a separate
+copy of the main training script for each process.
+
+In the single-machine synchronous case, `torch.distributed` or the
+:func:`torch.nn.parallel.DistributedDataParallel` wrapper may still have advantages over other
+approaches to data-parallelism, including :func:`torch.nn.DataParallel`:
+
+* Each process maintains its own optimizer and performs a complete optimization step with each
+  iteration. While this may appear redundant, since the gradients have already been gathered
+  together and averaged across processes and are thus the same for every process, this means
+  that no parameter broadcast step is needed, reducing time spent transferring tensors between
+  nodes.
+* Each process contains an independent Python interpreter, eliminating the extra interpreter
```

Comment:
Did you actually benchmark DDP vs DP? I'd be surprised if DDP turned out to be much faster, and it's much harder to use (you need to init distributed and launch all workers), so I don't think we should be advertising it for this use.